src/util/storageVolumeMigration.tsx

```diff
@@ -0,0 +1,182 @@
+import type { LxdStorageVolume } from "types/storage";
+import { capitalizeFirstLetter } from "./helpers";
+import { queryKeys } from "./queryKeys";
+import { hasLocation } from "./storageVolume";
+import VolumeLinkChip from "pages/storage/VolumeLinkChip";
+
+export type VolumeMigrationType = "cluster member" | "pool" | "project" | "";
+
+interface Props {
+  volume: LxdStorageVolume;
+  type: VolumeMigrationType;
+  target: string;
+  close: () => void;
+}
+
+export const useStorageVolumeMigration = ({
+  volume,
+  close,
+  type,
+  target,
+}: Props) => {
+  const toastNotify = useToastNotification();
+  const eventQueue = useEventQueue();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+
+  const handleSuccess = () => {
+    let successMessage: ReactNode = "";
+
+    const memberPath = hasLocation(volume)
+      ? `/member/${encodeURIComponent(volume.location)}`
+      : "";
+    const oldVolumeUrl = `/ui/project/${encodeURIComponent(volume.project)}/storage/pool/${encodeURIComponent(volume.pool)}${memberPath}/volumes/${encodeURIComponent(volume.type)}/${encodeURIComponent(volume.name)}`;
+    const newVolumeUrl = `/ui/project/${encodeURIComponent(volume.project)}/storage/pool/${encodeURIComponent(target)}${memberPath}/volumes/${encodeURIComponent(volume.type)}/${encodeURIComponent(volume.name)}`;
+
+    const volumeLink = (
+      <ResourceLink type="volume" value={volume.name} to={newVolumeUrl} />
+    );
+
+    const poolLink = (
+      <ResourceLink
+        type="pool"
+        value={target}
+        to={`/ui/project/${encodeURIComponent(volume.project)}/storage/pool/${encodeURIComponent(target)}`}
+      />
+    );
+
+    if (type === "cluster member") {
+      successMessage = (
+        <>
+          Volume <VolumeLinkChip volume={volume} /> successfully migrated to
+          cluster member{" "}
+          <ResourceLink
+            type="cluster-member"
+            value={target}
+            to={`/ui/cluster/member/${encodeURIComponent(target)}`}
+          />
+        </>
+      );
+    }
+
+    if (type === "pool") {
+      successMessage = (
+        <>
+          Volume {volumeLink} successfully migrated to pool {poolLink}
+        </>
+      );
+      if (oldVolumeUrl !== newVolumeUrl) {
+        navigate(newVolumeUrl);
+      }
+    }
+
+    if (type === "project") {
+      successMessage = (
+        <>
+          Instance <VolumeLinkChip volume={{ ...volume, project: target }} />
+          successfully moved to project{" "}
+          <ResourceLink
+            type="project"
+            value={target}
+            to={`/ui/project/${encodeURIComponent(target)}`}
+          />
+        </>
+      );
+
+      const oldUrl = window.location.pathname;
+      const newUrl = oldUrl.replace(
+        `/project/${volume.project}/instance/${volume.name}`,
+        `/project/${target}/instance/${volume.name}`,
+      );
+      if (oldUrl !== newUrl) {
+        navigate(newUrl);
+      }
```

Comment:
This does not update when migrating to a different project. We should consider the new project in `newUrl`